src/librustc_back/target/windows_msvc_base.rs

```diff
@@ -53,6 +53,7 @@ pub fn opts() -> TargetOptions {
         exe_suffix: ".exe".to_string(),
         staticlib_prefix: "".to_string(),
         staticlib_suffix: ".lib".to_string(),
+        no_default_libraries: false,
```

Comment:
The rumprun linker wrapper uses a GCC spec file to add a path to the rumprun version of system libraries (libc, libpthread etc). If `-nodefaultlibs` is specified, gcc will not use specified the path.

Maybe @mato has more insight to this, he helped me debug the issue.